### PR TITLE
feat(mods): metamod/amxmodx for goldsrc engines

### PR DESCRIPTION
**Pull request by tkrn**
# Description
The initial Metamod was renamed to Metamod: Source for accurate in-code representation. Metamod by itself refers to the HL1/goldsrc engines. Metamod along with amxmodx (base) and game specific versions were added for supported goldsrc engines. It also modifies required files liblist.gam and plugin.ini for Metamod. The majority of the testing has been done against csserver but all the other varients such as dod, tfc, ns, and ts also ahead to the exact same logic. Added logic for dependencies (metamod, amxmodx) while installing. The only to-do item *maybe* is to add dependency checks while removing/updating a mod. The likely hood of these mods being updated is slim given the age of the game/mods.

Fulfils request of feature (issue #2256)

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [X] New feature (change which adds functionality).
* [ ] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [X] This pull request links to an issue.
* [X] This pull request uses the `develop` branch as its base.
* [X] This pull request Subject follows the Conventional Commits standard.
* [X] This code follows the style guidelines of this project.
* [X] I have performed a self-review of my code.
* [X] I have checked that this code is commented where required.
* [X] I have provided a detailed with enough description of this PR.
* [X] I have checked If documentation needs updating.

## Documentation

This will need to be updated after it is tested/merged - https://docs.linuxgsm.com/commands/mods#supported-mods-add-ons

### CODE REVIEW

File: lgsm/functions/mods_list.sh [hunk 1]
```diff
@@ -93,11 +154,14 @@ mod_info_wiremodextras=( MOD "wiremod-extras" "Wiremod Extras" "https://github.c
...
 mod_info_sdtdoxide=( MOD "sdtdoxide" "Oxide for 7 Days To Die" "${oxidesdtdlatestlink}" "Oxide.SevenDaysToDie.zip" "0" "LowercaseOff" "${systemdir}" "OVERWRITE" "ENGINES" "7 Days To Die;" "NOTGAMES" "https://umod.org/games/7-days-to-die" "Allows for the use of plugins" )
 
 # REQUIRED: Set all mods info into the global array
+<<<<<<< HEAD
 mods_global_array=( "${mod_info_metamod[@]}" "${mod_info_sourcemod[@]}" "${mod_info_steamworks[@]}" "${mod_info_stripper[@]}" "${mod_info_gokz[@]}" "${mod_info_ttt[@]}" "${mod_info_get5[@]}" "${mod_info_pug[@]}" "${mod_info_ulib[@]}" "${mod_info_ulx[@]}" "${mod_info_utime[@]}" "${mod_info_uclip[@]}" "${mod_info_acf[@]}" "${mod_info_acf_missiles[@]}" "${mod_info_acf_sweps[@]}" "${mod_info_advdupe2[@]}" "${mod_info_pac3[@]}" "${mod_info_wiremod[@]}" "${mod_info_wiremodextras[@]}" "${mod_info_darkrp[@]}" "${mod_info_darkrpmodification[@]}" "${mod_info_rustoxide[@]}" "${mod_info_hwoxide[@]}" "${mod_info_sdtdoxide[@]}" )
+=======
```
Comment (dgibbs64) [inline lgsm/functions/mods_list.sh:165]: Remove this

File: lgsm/functions/mods_list.sh [hunk 2]
```diff
@@ -93,11 +154,14 @@ mod_info_wiremodextras=( MOD "wiremod-extras" "Wiremod Extras" "https://github.c
...
+mods_global_array=( "${mod_info_metamod[@]}" "${mod_info_base_amxx[@]}" "${mod_info_cs_amxx[@]}" "${mod_info_dod_amxx[@]}" "${mod_info_tfc_amxx[@]}" "${mod_info_ns_amxx[@]}" "${mod_info_ts_amxx[@]}" "${mod_info_metamodsource[@]}" "${mod_info_sourcemod[@]}" "${mod_info_steamworks[@]}" "${mod_info_gokz[@]}" "${mod_info_ttt[@]}" "${mod_info_get5[@]}" "${mod_info_pug[@]}" "${mod_info_ulib[@]}" "${mod_info_ulx[@]}" "${mod_info_utime[@]}" "${mod_info_uclip[@]}" "${mod_info_acf[@]}" "${mod_info_acf_missiles[@]}" "${mod_info_acf_sweps[@]}" "${mod_info_advdupe2[@]}" "${mod_info_pac3[@]}" "${mod_info_wiremod[@]}" "${mod_info_wiremodextras[@]}" "${mod_info_darkrp[@]}" "${mod_info_darkrpmodification[@]}" "${mod_info_rustoxide[@]}" "${mod_info_hwoxide[@]}" "${mod_info_sdtdoxide[@]}" )
+>>>>>>> first release of metamod/amxmodx for goldsrc/hl1 engines
```
Comment (dgibbs64) [inline lgsm/functions/mods_list.sh:167]: remove this

File: lgsm/functions/mods_list.sh [hunk 3]
```diff
@@ -93,11 +154,14 @@ mod_info_wiremodextras=( MOD "wiremod-extras" "Wiremod Extras" "https://github.c
...
 mod_info_darkrpmodification=( MOD "darkrpmodification" "DarkRP Modification" "https://github.com/FPtje/darkrpmodification/archive/master.zip" "darkrpmodification-master.zip" "0" "LowercaseOff" "${systemdir}/addons" "NOUPDATE" "ENGINES" "Garry's Mod;" "NOTGAMES" "http://darkrp.com" "Customize DarkRP settings" )
 
-
 # Oxidemod
 mod_info_rustoxide=( MOD "rustoxide" "Oxide for Rust" "${oxiderustlatestlink}" "Oxide.Rust-linux.zip" "0" "LowercaseOff" "${systemdir}" "OVERWRITE" "ENGINES" "Rust;" "NOTGAMES" "https://umod.org/games/rust" "Allows for the use of plugins" )
 mod_info_hwoxide=( MOD "hwoxide" "Oxide for Hurtworld" "${oxidehurtworldlatestlink}" "Oxide.Hurtworld.zip" "0" "LowercaseOff" "${systemdir}" "OVERWRITE" "ENGINES" "Hurtworld;" "NOTGAMES" "https://umod.org/games/hurtworld" "Allows for the use of plugins" )
 mod_info_sdtdoxide=( MOD "sdtdoxide" "Oxide for 7 Days To Die" "${oxidesdtdlatestlink}" "Oxide.SevenDaysToDie.zip" "0" "LowercaseOff" "${systemdir}" "OVERWRITE" "ENGINES" "7 Days To Die;" "NOTGAMES" "https://umod.org/games/7-days-to-die" "Allows for the use of plugins" )
 
 # REQUIRED: Set all mods info into the global array
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (dgibbs64) [reply lgsm/functions/mods_list.sh:163]: This is a merge compare

**Review by dgibbs64 — CHANGES_REQUESTED** [review]
Skimming over the code it looks good. I would like a volunteer to test the code and leave a review

**Comment by tkrn** [comment]
Removed the accidental merge compare in commit d1d0f81.

**Review by dgibbs64 — DISMISSED** [review]
Ran a test and looks like it works 👍

**Comment by tkrn** [comment]
See pull request for documentation update - https://github.com/GameServerManagers/LinuxGSM-Docs/pull/50

**Pull request merged**